client/src/components/pages/wrapper-columns/participants-board/api/useAllParticipants.ts

```diff
@@ -0,0 +1,79 @@
+import { useEffect, useState } from 'react';
+
+import { StateProps } from '@/types';
+import { useUserInfo } from '@/context';
+import { normalizeUsers } from '@/utils';
+import { showMessage } from '@/components';
+import { applyUserToBoardAction, getAllUsersByOrganizationIdActions } from '@/actions';
+
+import { sortedUsers } from './helper';
+import { UserParticipants } from './types';
+
+interface IAddUserToBoard {
+  applyUserId: string;
+  setBoardUser: StateProps<UserParticipants[]>;
+}
+
+export const useAllParticipants = (boardId: string, boardUsers: UserParticipants[]) => {
+  const { _id, organizationId } = useUserInfo();
+  const [isLoading, setIsLoading] = useState(true);
+  const [participants, setAllParticipants] = useState<UserParticipants[]>([]);
+
+  const getParticipants = async () => {
+    try {
+      setIsLoading(true);
+      const response = await getAllUsersByOrganizationIdActions({
+        id: organizationId ? String(organizationId) : '',
+        page: 1,
+        limit: 200,
+      });
+
+      if (response && response?.success && response?.users) {
+        const users = JSON.parse(response.users);
+
+        const allUsers = normalizeUsers(users.results);
+
+        setAllParticipants(
+          sortedUsers(allUsers.filter((user) => !boardUsers.find((boardUser) => boardUser.id === user.id))),
```

Comment:
would you mind extracting all that filter logic into a separate constant with an appropriate name?